docs/content/basic/assistant.md

```diff
@@ -0,0 +1,104 @@
+    await saveThreadContext();
+  },
+  userMessage: async ({ client, message, say, setTitle, setStatus }) => {
+    const { channel, thread_ts } = message;
+
+    // Set the title of the Assistant thread to capture the initial topic/question
+    await setTitle(message.text);
+
+    // Set the status of the Assistant to give the appearance of active processing.
+    await setStatus('is typing..');
+
+    // Retrieve the Assistant thread history for context of question being asked
+    const thread = await client.conversations.replies({
+      channel,
+      ts: thread_ts,
+      oldest: thread_ts,
+    });
+
+    // Prepare and tag each message for LLM processing
+    const userMessage = { role: 'user', content: message.text };
+    const threadHistory = thread.messages.map((m) => {
+      const role = m.bot_id ? 'assistant' : 'user';
+      return { role, content: m.text };
+    });
+
+    const messages = [
+      { role: 'system', content: DEFAULT_SYSTEM_CONTENT },
+      ...threadHistory,
+      userMessage,
+    ];
+
+    // Send message history and newest question to LLM
+    const llmResponse = await openai.chat.completions.create({
+      model: 'gpt-4o-mini',
+      n: 1,
+      messages,
+    });
+
+    // Provide a response to the user
+    await say(llmResponse.choices[0].message.content);
+  },
+});
+
+app.assistant(assistant);
+```
+
+When a user opens an Assistant thread while in a channel, the channel information is stored as the thread's `AssistantThreadContext` data. You can access this information by using the `getThreadContext()` utility. The reason Bolt provides this utility is that the most recent thread context information is not included in the subsequent user message event payload data. Therefore, an app must store the context data when it is changed so that the app can refer to the data in message event listeners.
+
+When the user switches channels, the `assistant_thread_context_changed` event will be sent to your app. If you use the built-in `AssistantThreadContextStore` without any custom configuration (like the above code snippet does), the updated context data is automatically saved as message metadata on the first reply from the assistant bot.
+
+As long as you use the built-in approach, you don't need to store the context data within a datastore. The downside of this default behavior is the overhead of additional calls to the Slack API. These calls include those to `conversations.history` which are used to look up the stored message metadata that contains the thread context (via `getThreadContext()`).
+
+To store context elsewhere, pass a custom `AssistantThreadContextStore` implementation to the `Assistant` constructor through the `threadContextStore` property. If not provided, Bolt utilizes a `DefaultThreadContextStore`, which is a reference implementation that relies on storing and retrieving message metadata as the context changes. A custom `threadContextStore` must feature `get` and `save` methods.
```

Comment:
The default one is good to go for production apps! Only when a developer has to minimize Slack API calls in a large coporation workspaces, they may want to avoid relying on the default one.

Reply:
Ah, good to know! Here I am spreading misinformation and underselling y/our good work. 😅  Will adjust it accordingly.